packages/pdf-renderer/src/lib/pdfGenerator.service.ts

```diff
@@ -40,23 +44,50 @@ export class PdfGenerator {
         };
     }
 
-    private static generatePdfBufferTask: TaskFunction<GeneratePageParams, Buffer> = async ({ page, data }) => {
+    private static async preparePageForScreenshot(page: Page, { html }: GeneratePageScreenshotParams): Promise<void> {
+        await page.setContent(html, {
+            timeout: 60000,
+            waitUntil: ["load", "domcontentloaded"],
+        });
+    }
+
+    private static generatePdfBufferTask: TaskFunction<GeneratePdfPageParams, Buffer> = async ({ page, data }) => {
         const pdfOptions = await PdfGenerator.preparePage(page, data);
 
         return await page.pdf(pdfOptions);
     };
 
-    private static generatePdfStreamTask: TaskFunction<GeneratePageParams, Readable> = async ({ page, data }) => {
+    private static generatePdfStreamTask: TaskFunction<GeneratePdfPageParams, Readable> = async ({ page, data }) => {
         const pdfOptions = await PdfGenerator.preparePage(page, data);
 
         return await page.createPDFStream(pdfOptions);
     };
 
-    async generateBuffer(params: GeneratePageParams) {
+    private static generateScreenshotTask: TaskFunction<GeneratePageScreenshotParams, Buffer> = async ({
+        page,
+        data,
+    }) => {
+        await PdfGenerator.preparePageForScreenshot(page, data);
+
+        return (await page.screenshot({
+            captureBeyondViewport: data.captureBeyondViewport,
+            type: data.type,
+            fullPage: data.fullPage,
+            quality: data.quality,
+            clip: data.clip,
+            fromSurface: data.fromSurface,
```

Comment:
Hmm, I don't think it would be better but I think you could destructure `data` by doing `const {html, ...screenshotOptions} = data` and then pass `screenshotOptions` as a `page.screenshot` param. By doing that we wouldn't have to worry about forgetting to pass one of the options. Alternatively I think you could even just pass `data` and don't worry that it contains `html` field. What do you think?